File: playground-golang/examples/n206/n206.go
package n206

import (
	"drremain.cn/playground-golang/DataStructure"
)

type ListNode = DataStructure.ListNode

func reverseList(head *ListNode) *ListNode {
	var prev *ListNode
	node := head
	for node != nil {
		tmp := node.Next
		node.Next = prev
		prev = node
		node = tmp
	}
	return prev
}

File: playground-python/src/releetcode/problems/n206.py
from typing import Optional
from ..data_structures.list_node import ListNode

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        prev = None
        node = head
        while node is not None:
            tmp = node.next
            node.next = prev
            prev = node
            node = tmp
        return prev

